libfwupdplugin/fu-plugin.c

```diff
@@ -357,17 +352,30 @@ fu_plugin_open(FuPlugin *self, const gchar *filename, GError **error)
 		return FALSE;
 	}
 
+	/* call the new-style fu_plugin_init_vfuncs() vfunc setup */
+	g_module_symbol(priv->module, "fu_plugin_init_vfuncs", (gpointer *)&init_vfuncs);
+	if (init_vfuncs == NULL) {
+		g_set_error(error,
+			    G_IO_ERROR,
+			    G_IO_ERROR_FAILED,
+			    "failed to init_vfuncs() on plugin %s",
+			    filename);
+		fu_plugin_add_flag(self, FWUPD_PLUGIN_FLAG_FAILED_OPEN);
+		fu_plugin_add_flag(self, FWUPD_PLUGIN_FLAG_USER_WARNING);
+		return FALSE;
+	}
+	init_vfuncs(vfuncs);
+
 	/* set automatically */
 	if (fu_plugin_get_name(self) == NULL) {
 		g_autofree gchar *str = fu_plugin_guess_name_from_fn(filename);
 		fu_plugin_set_name(self, str);
 	}
 
 	/* optional */
-	g_module_symbol(priv->module, "fu_plugin_init", (gpointer *)&func);
-	if (func != NULL) {
+	if (vfuncs->init != NULL) {
```

Comment:
this should be compulsory no?

Reply:
With the super new `vfuncs->build_hash`, no.